### PR TITLE
Support detection of debugger commands with arguments

### PR DESCRIPTION
**Pull request by crazydog115**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,9 @@
 * Make it possible to disable `Lint/UnneededDisable`. ([@jonas054][])
 * [#1958](https://github.com/bbatsov/rubocop/issues/1958): Show name of `Lint/UnneededDisable` when `-D/--display-cop-names` is given. ([@jonas054][])
 
+### New features
```
Comment (bbatsov) [inline CHANGELOG.md]: This section should be before `Bugs fixed`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,9 @@
...
+* `Debugger` cop now checks catches methods called with arguments. ([@crazydog115][])
```
Comment (bbatsov) [inline CHANGELOG.md]: Add a blank line after the title.

File: spec/rubocop/cop/lint/debugger_spec.rb [hunk 3]
```diff
@@ -50,6 +50,51 @@
                 save_screenshot))
   end
 
+  it 'reports an offense for a debugger call with an argument' do
```
Comment (bbatsov) [inline spec/rubocop/cop/lint/debugger_spec.rb]: There's a bit of duplication in the specs for this cop. Maybe you can introduce some shared examples to reduce it.

**Comment by bbatsov** [comment]
Which of those methods actually accept any params?

**Comment by bquorning** [comment]
`byebug` did accept params until v4.0.4: https://github.com/deivid-rodriguez/byebug/blob/v4.0.3/lib/byebug/attacher.rb#L25

`debugger` still does: https://github.com/cldwalker/debugger/blob/10a25b70c3466f79db95f361ed34c0e7da1551e6/lib/ruby-debug.rb#L169 (and that method is aliased as `#breakpoint` which we should probably check for as well.

All of the Capybara methods accept a path parameter: https://github.com/jnicklas/capybara/blob/3ae284460b1af35d40b077bf14f7222c2982c120/lib/capybara/session.rb#L643

Don’t know about Pry.

**Comment by crazydog115** [comment]
I'm not sure if any of the `pry` commands support arguments, but the
cop is designed to ignore any arguments. So whether or not the command
supports arguments, or if it ever does, does not matter. If the
command at its root is in the code, it should be caught by the cop, in
my opinion.

File: spec/rubocop/cop/lint/debugger_spec.rb [hunk 4]
```diff
@@ -6,73 +6,77 @@
...
-      src = "code.#{method_name}"
-      inspect_source(cop, src)
-      expect(cop.offenses).to be_empty
+      wont_report_offence_for "code.#{method_name}"
     end
   end
+
+  private
+
+  def should_report_offence_for(src)
```
Comment (bbatsov) [inline spec/rubocop/cop/lint/debugger_spec.rb]: Why don't you use RSpec's shared examples for this?

Comment (crazydog115) [inline spec/rubocop/cop/lint/debugger_spec.rb]: Sorry, I'm not that familiar with RSpec. I'll look into it! :)

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**